onnxruntime/python/onnxruntime_pybind_mlvalue.cc

```diff
@@ -194,9 +194,15 @@ bool PyObjectCheck_Array(PyObject* o) {
   return PyObject_HasAttrString(o, "__array_finalize__");
 }
 
+void CpuToCpuMemCpy(void* dst, const void* src, size_t num_bytes) {
+  memcpy(dst, src, num_bytes);
+}
+
 // Expects p_tensor properly created
 // Does not manage darray life-cycle
-void CopyDataToTensor(PyArrayObject* darray, int npy_type, std::unique_ptr<Tensor>& p_tensor) {
+
+void CopyDataToTensor(PyArrayObject* darray, int npy_type, std::unique_ptr<Tensor>& p_tensor,
+                      MemCpyFunc mem_cpy_to_device = CpuToCpuMemCpy) {
```

Comment:
>CpuToCpuMemCpy [](start = 53, length = 14)

Will we always invoke copying? Why would we need CpuToCpu?

Reply:
This is a helper method that already existed that is mostly invoked for string tensors. However, it does support copying data buffers too - if the user were to call this method for non-string tensors. If you scrolled down, you'd see that it uses std::memcpy() which is essentially `CpuToCpuMemCpy`. All I did is add a mechanism for callers to define their own memcpy function and the default value for the parameter is basically std::memcpy which basically means status quo is maintained.